### PR TITLE
fix acf blocks unicode (WP-500)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/GutenbergBlockHelper.php [hunk 1]
```diff
@@ -8,11 +8,6 @@
 use Smartling\Exception\SmartlingGutenbergParserNotFoundException;
 use Smartling\Helpers\EventParameters\TranslationStringFilterParameters;
 
-/**
- * Class SubstringProcessorHelperAbstract
- *
- * @package Smartling\Helpers
- */
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/GutenbergBlockHelper.php]: Unrelated change

File: inc/Smartling/Helpers/GutenbergBlockHelper.php [hunk 2]
```diff
@@ -195,24 +190,24 @@ public function processString(TranslationStringFilterParameters $params)
...
+	public function parseBlocks( $string ) {
+		if ( function_exists( '\parse_blocks' ) ) {
+			return \parse_blocks( $string );
+		}
+		if ( function_exists( '\gutenberg_parse_blocks' ) ) {
+			return \gutenberg_parse_blocks( $string );
+		}
+
+		throw new SmartlingGutenbergParserNotFoundException( 'No block parser found.' );
+	}
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/GutenbergBlockHelper.php]: Could you please set up code style rules for your IDE (replicate that ones which are already in connector code). It's hard to review such things

File: tests/Smartling/Helpers/GutenbergBlockHelperTest.php [hunk 3]
```diff
@@ -265,15 +265,32 @@ public function placeBlockDataProvider()
...
      */
     public function testRenderGutenbergBlock($blockName, array $attributes, array $chunks, $expected)
     {
-        $result = $this->invokeMethod($this->helper, 'renderGutenbergBlock', [$blockName, $attributes, $chunks]);
-        self::assertEquals($expected, $result);
-    }
+		self::assertEquals($expected, $this->helper->renderGutenbergBlock($blockName, $attributes, $chunks));
+	}
+
+	public function testIssueWP500() {
```
Comment (dimitrystd) [inline tests/Smartling/Helpers/GutenbergBlockHelperTest.php]: Let's use meaningful names. It will be hard to understand what's affected when you see a list of failed tests in jenkins
An example `testRenderGutenbergBlockWithAccentedCharacters`. Don't afraid of long test names.

Comment (dimitrystd) [inline tests/Smartling/Helpers/GutenbergBlockHelperTest.php]: May i propose one more unit test for emoji and other utf8mb4 characters. Pls add a test like `testRenderGutenbergBlockWithEmoji` and use test string  
```
Test 𝒞 and 😂, 絵文字, 👩‍🦽, ⚛️.
```

File: tests/Smartling/Helpers/GutenbergBlockHelperTest.php [hunk 4]
```diff
@@ -265,15 +265,32 @@ public function placeBlockDataProvider()
...
+					'cta_copy'        => 'Obter uma cotação',
+					'cta_url'         => 'https://www.wework.com/host-event-contact',
+					'sticky_behavior' => 'bottom',
+				],
+			'align' => '',
+			'mode'  => 'auto',
+		], [] );
+		$this->assertEquals( '<!-- wp:acf/sticky-cta {"id":"block_5e46fa29a5a8e","name":"acf\/sticky-cta",' .
+		                     '"data":{"copy":"Pronto para reservar seu próximo evento?","cta_copy":"Obter uma cotação"' .
+		                     ',"cta_url":"https:\/\/www.wework.com\/host-event-contact","sticky_behavior":"bottom"},' .
```
Comment (dimitrystd) [inline tests/Smartling/Helpers/GutenbergBlockHelperTest.php]: `www.wework.com` - let's remove from publicly available source code

File: readme.txt [hunk 5]
```diff
@@ -57,6 +57,9 @@ Additional information on the Smartling Connector for WordPress can be found [he
 3. Track translation status within WordPress from the Submissions Board. View overall progress of submitted translation requests as well as resend updated content.
 
 == Changelog ==
+= 1.12.1 =
+* Fixed compatibility with ACF plugin and non-latin characters in translation.
```
Comment (dimitrystd) [inline readme.txt]: Below proposed text for 2 recent releases. The current one is too short
```
= 1.12.1 =
* Changed default behavior of how we handle related content for "request translations" flow (see v1.12.0 for more details):
 * When you request translation from `Post Edit` screen it submits only current page and optionally you can tell to submit only immediate related content (tags, categories, images, etc)
 * When you request from `Bulk Submit` screen then it will submit only chosen content (any related content should be requested manually)
* Fixed compatibility with ACF plugin and extended unicode characters (emoji, accented characters) in translation.
```
I intentionally repeated in 1.12.1 how it will work despite of description in 1.12.0.



```
= 1.12.0 =
This release brings major changes how the connector works with related content. An example, a page may have a tag and featured image. Before when you requested translation for a page the connector submitted page itself plus tag and image (3 file uploads in total). This approach guaranteed that the translated page will look similar to the original page and layout is not broken. The drawback of this approach, you can't control how much content will be uploaded to Smartling. If your site has cross-references between pages then requesting a single page may trigger hundreds of uploads.
The new connector version adds the new toggle that controls this behavior. Now you can tell the connector to submit only requested content or requested content + immediate related content.

The new toggle can be found `Smartling` -> `Settings` -> `Show Expert Settings` -> `Handle relations`, where:

* `Automatically` - use legacy approach, upload to smartling as much as possible related content
* `Manually` - use new approach, upload only requested content or requested content + immediate related content
```
